### PR TITLE
[chip-tool] Interactive mode does not work great with complex/custom …

### PR DESCRIPTION
**Pull request by vivien-apple**
…arguments using json formatting

#### Problem

In #19238 "double" quotes supports has been added to the interactive mode in order to pass string that may contains spaces instead of them being splitter into 2 arguments.

In this PR I would like to replace double quotes by double quote does not works great when one is trying to copy-paste complex json arguments to the command line such as:

`$ chip-tool accesscontrol write acl '[{"fabricIndex": 1, "privilege": 5, "authMode": 2, "subjects": [112233,1111], "targets":null},{"fabricIndex": 1, "privilege": 3, "authMode": 3, "subjects": [3333], "targets":null}]' 0x12344321 0`

One would need to escape all the `"` characters to make it works properly, while using single quotes means such strings can be copy pasted without escaping and if one wants to pass an SSID that contains space, it should be 'My SSID' in stead of "My SSSID" which is a much easier and readable change.

#### Change overview
 * Use single quote character as the delimiter for std::quoted and not double quotes.

#### Testing

I tried to convert use the previous write all command into interactive mode.

### CODE REVIEW

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]
Fast tracking tooling

File: examples/chip-tool/commands/interactive/InteractiveCommands.cpp [hunk 1]
```diff
@@ -99,7 +99,7 @@ bool InteractiveStartCommand::ParseCommand(char * command)
     std::string arg;
 
     std::stringstream ss(command);
-    while (ss >> std::quoted(arg))
+    while (ss >> std::quoted(arg, '\''))
```
Comment (krypton36) [inline examples/chip-tool/commands/interactive/InteractiveCommands.cpp:102]: Can you do this for darwin-framework-tool as well please?

**Review by krypton36 — CHANGES_REQUESTED** [review]

**Review by krypton36 — APPROVED** [review]

**Comment by vivien-apple** [comment]
> Why not just optionally accept a data model payload argument at stdin?

Do you mean an hex encoded data payload ?

**Comment by msandstedt** [comment]
I'm not suggesting any specific format.  I'm just pointing out that if we are struggling with escape sequences in the command line interpreter, stdin is available to receive anything we like.

**Pull request merged**